src/main/java/dd/datetimehandler/DateTimeHandler.java

```diff
@@ -0,0 +1,119 @@
+package dd.datetimehandler;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import dd.tasks.Deadline;
+import dd.tasks.Event;
+import dd.tasks.Task;
+
+/**
+ * A date time handler checks if a date input given is valid and
+ * parses them according to the given format.
+ */
+public class DateTimeHandler {
+
+    /**
+     * Returns whether input is a valid date or date and time input.
+     *
+     * @param input String containing input date or date and time.
+     * @return true if input is valid, false if not.
+     */
+    public boolean isValidInput(String input) {
+        if (input.length() == 10) {
+            try {
+                convertDate(input);
+                return true;
+            } catch (DateTimeParseException e) {
+                return false;
+            }
+        } else if (input.length() == 15) {
+            try {
+                convertDateTime(input);
+                return true;
+            } catch (DateTimeParseException e) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns the appropriate method based on input given.
```

Comment:
While it can be seen what this method does from @param and @return, perhaps a little more information on that in the description will be helpful?